### PR TITLE
Bug 1868792 - Add zip file mime type to Fenix the FilePicker.

### PR DESCRIPTION
**Pull request by Amejia481**
Needs to land first https://phabricator.services.mozilla.com/D195594

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1868792

### CODE REVIEW

File: android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonFilePicker.kt [hunk 1]
```diff
@@ -73,6 +73,6 @@ class AddonFilePicker(
 internal open class AddonOpenDocument : ActivityResultContracts.OpenDocument() {
     override fun createIntent(context: Context, input: Array<String>): Intent {
         // We're restricting the mime type to XPI files, because that's expected by GeckoView currently.
```
Comment (willdurand) [inline android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonFilePicker.kt:75]: I suppose we can remove this comment now.

Comment (Amejia481) [reply android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonFilePicker.kt:75]: 😅

Comment (Amejia481) [reply android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonFilePicker.kt]: Updated!

File: android-components/components/feature/addons/src/test/java/mozilla/components/feature/addons/ui/AddonFilePickerTest.kt [hunk 2]
```diff
@@ -85,4 +86,14 @@ class AddonFilePickerTest {
...
     }
+
+    @Test
+    fun `WHEN creating an intent from a AddonOpenDocument THEN it contains mime types for zips and xpi files`() {
+        val contractResult = AddonOpenDocument()
+
+        val intent = contractResult.createIntent(testContext, arrayOf())
+
+        val mimeTypes = intent.extras!!.getStringArray(Intent.EXTRA_MIME_TYPES)
+        assertTrue(mimeTypes!!.all { it!! == "application/x-xpinstall" || it == "application/zip" })
```
Comment (willdurand) [inline android-components/components/feature/addons/src/test/java/mozilla/components/feature/addons/ui/AddonFilePickerTest.kt:97]: This assertion will still pass if we remove one of the two mime types in `AddonFilePicker`, that isn't great. Let's verify the whole list of mime types instead:

```suggestion
        assertArrayEquals(mimeTypes, arrayOf("application/x-xpinstall", "application/zip"))
```

**Review by willdurand — APPROVED** [review]
r+wc, thanks!

File: android-components/components/feature/addons/src/test/java/mozilla/components/feature/addons/ui/AddonFilePickerTest.kt (hunk 2 above)
Comment (Amejia481) [reply android-components/components/feature/addons/src/test/java/mozilla/components/feature/addons/ui/AddonFilePickerTest.kt]: Thanks for the suggestion, I updated the PR with it :)

**Pull request merged**